Notation.Plugin.AzureKeyVault/KeyVault/KeyVaultClient.cs

```diff
@@ -0,0 +1,149 @@
+        }
+
+        /// <summary>
+        /// Constructor to create AzureKeyVault object from key identifier or
+        /// certificate identifier.
+        ///
+        /// <param name="id">
+        /// Key identifier or certificate identifier. (e.g. https://<vaultname>.vault.azure.net/keys/<name>/<version>)
+        /// </param>
+        /// </summary>
+        public KeyVaultClient(string id) : this(ParseId(id)) { }
+
+        /// <summary>
+        /// A helper constructor to create KeyVaultClient from KeyVaultMetadata.
+        /// </summary>
+        private KeyVaultClient(KeyVaultMetadata metadata)
+            : this(metadata.KeyVaultUrl, metadata.Name, metadata.Version) { }
+
+        /// <summary>
+        /// A helper function to parse key identifier or certificate identifier
+        /// and return KeyVaultMetadata.
+        /// </summary>
+        private static KeyVaultMetadata ParseId(string id)
+        {
+            if (string.IsNullOrEmpty(id))
+            {
+                throw new ArgumentNullException(nameof(id), "Id must not be null or empty");
+            }
+
+            var uri = new Uri(id);
+            // Validate uri
+            if (uri.Segments.Length != 4)
+            {
+                throw new ValidationException(INVALID_INPUT_ERROR_MSG);
+            }
+
+            if (uri.Segments[1] != "keys/" && uri.Segments[1] != "certificates/")
+            {
+                throw new ValidationException(INVALID_INPUT_ERROR_MSG);
+            }
+
+            if (uri.Scheme != "https")
+            {
+                throw new ValidationException(INVALID_INPUT_ERROR_MSG);
+            }
+
+            return new KeyVaultMetadata(
+                KeyVaultUrl: $"{uri.Scheme}://{uri.Host}",
+                Name: uri.Segments[2].TrimEnd('/'),
+                Version: uri.Segments[3].TrimEnd('/')
+            );
+        }
+
+        /// <summary>
+        /// Sign the payload and return the signature.
+        /// </summary>
+        public async Task<byte[]> Sign(SignatureAlgorithm algorithm, byte[] payload)
+        {
+            var signResult = await _cryptoClient.Value.SignDataAsync(algorithm, payload);
+            if (signResult.KeyId != _keyId)
+            {
+                throw new PluginException($"Invalid keys identifier. The user provides {_keyId} but the response contains {signResult.KeyId} as the keys");
+            }
+
+            if (signResult.Algorithm != algorithm)
+            {
+                throw new PluginException($"Invalid signature algorithm. The user provides {algorithm} but the response contains {signResult.Algorithm} as the algorithm");
+            }
+
+            return signResult.Signature;
+        }
+
+        /// <summary>
+        /// Get the certificate from the key vault.
+        /// </summary>
+        public async Task<X509Certificate2> GetCertificate()
```

Comment:
Question: Should we follow naming conversion?
```suggestion
        public async Task<X509Certificate2> GetCertificateAsync()
```